merino/newtab/upday_provider.py

```diff
@@ -0,0 +1,95 @@
+"""Upday specific data provider for the New Tab."""
+import logging
+from typing import Any, Tuple
+
+from httpx import AsyncClient, BasicAuth, HTTPError
+
+from merino.newtab.base import Recommendation
+
+# The path to the Authentication Endpoint of the Upday API
+AUTH_URL_PATH = "/v1/oauth/token"
+LOCAL_ARTICLES_URL_PATH = "/v1/ntk/articles"
+
+logger = logging.getLogger(__name__)
+
+
+class UpdayError(Exception):
+    """Errors related to calling the Upday API."""
+
+    pass
+
+
+class UpdayProvider:
+    """Provider for getting Upday Information."""
+
+    def __init__(self, username: str, password: str, http_client: AsyncClient):
+        """Initialize the Upday Provider to get information."""
+        self.basic_auth = BasicAuth(username=username, password=password)
+        self.http_client = http_client
+
+    async def get_upday_recommendations(
+        self, locale: str, language: str
+    ) -> list[Recommendation]:
+        """Get the Upday recommendations given the local and language of the request."""
+        access_token, token_type = await self._get_access_token()
+
+        articles = await self._get_articles_from_upday(
+            access_token, language, locale, token_type
+        )
+
+        recommendations = []
+        for article in articles:
+            url_to_use: str = (
+                article["partnerUrl"]
+                if article.get("partnerUrl")
+                else article.get("url")
+            )
+            recommendations.append(
+                Recommendation(
+                    url=url_to_use,
```

Comment:
nit: you can drop the block for `url_to_use`
```suggestion
                    url=article["partnerUrl"] or article["url"],
```